status code 500 returned on trying to use non existing CI backend

Then trying to call POST on /api/submitjob/<group>/<project> with the following POST body:

```
data = {
    "definition": <definition>,
    "backend": <backend-name>
}

```

When backend-name is not defined in the instance, following exception happens:

```
[2017-05-03 13:26:23 +0000] [ERROR] Internal Server Error: /api/submitjob/lite/zephyr-upstream/ced1a8bacf6e03ec078c047db93d46a791a91c39/arduino101
Traceback (most recent call last):
  File "/home/milosz/linaro/tcstorm/squad/squad-venv/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/home/milosz/linaro/tcstorm/squad/squad-venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/home/milosz/linaro/tcstorm/squad/squad-venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/milosz/linaro/tcstorm/squad/squad-venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/milosz/linaro/tcstorm/squad/squad-venv/lib/python3.5/site-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/home/milosz/linaro/tcstorm/squad/squad-venv/lib/python3.5/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/milosz/linaro/tcstorm/squad/squad/http.py", line 25, in auth_wrapper
    return func(*args, **kwargs)
  File "/home/milosz/linaro/tcstorm/squad/squad/api/ci.py", line 17, in submit_job
    backend = Backend.objects.get(name=request.POST.get('backend'))
  File "/home/milosz/linaro/tcstorm/squad/squad-venv/lib/python3.5/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/milosz/linaro/tcstorm/squad/squad-venv/lib/python3.5/site-packages/django/db/models/query.py", line 385, in get
    self.model._meta.object_name
squad.ci.models.DoesNotExist: Backend matching query does not exist.
```

I guess some other status code should be returned instead of 500 (possibly with body describing the issue).

--- a/squad/api/ci.py
+++ b/squad/api/ci.py
@@ -1,4 +1,4 @@
-from django.http import HttpResponse
+from django.http import HttpResponse, HttpResponseBadRequest
 from django.views.decorators.csrf import csrf_exempt
 from django.views.decorators.http import require_http_methods
 
@@ -14,14 +14,29 @@ from squad.core.models import Project
 @csrf_exempt
 @auth
 def submit_job(request, group_slug, project_slug, version, environment_slug):
-    backend = Backend.objects.get(name=request.POST.get('backend'))
+    backend_name = request.POST.get('backend')
+    if backend_name is None:
+        return HttpResponseBadRequest("backend field is required")
+    backend = None
+    try:
+        backend = Backend.objects.get(name=request.POST.get('backend'))
+    except Backend.DoesNotExist:
+        return HttpResponseBadRequest("requested backend does not exist")
+
+    # project has to exist or request will result with 404
     project = Project.objects.get(slug=project_slug, group__slug=group_slug)
+    if backend is None or project is None:
+        return HttpResponseBadRequest("malformed request")
 
     # definition can be received as a file upload or as a POST parameter
+    definition = None
     if 'definition' in request.FILES:
         definition = read_file_upload(request.FILES['definition'])
     else:
         definition = request.POST.get('definition')
+
+    if definition is None:
+        return HttpResponseBadRequest("test job definition is required")
 
     # create TestJob object
     test_job = TestJob.objects.create(
@@ -34,4 +49,5 @@ def submit_job(request, group_slug, project_slug, version, environment_slug):
     # schedule submission
     submit.delay(test_job.id)
 
-    return HttpResponse('', status=201)
+    # return ID of test job
+    return HttpResponse(test_job.id, status=201)

--- a/test/api/test_ci.py
+++ b/test/api/test_ci.py
@@ -38,6 +38,21 @@ class CiApiTest(TestCase):
             ).count()
         )
 
+    def test_invalid_backend_test_run(self):
+        args = {
+            'backend': 'lava.foo',
+            'definition': 'foo: 1',
+        }
+        r = self.client.post('/api/submitjob/mygroup/myproject/1/myenv', args)
+        self.assertEqual(400, r.status_code)
+
+    def test_missing_definition_test_run(self):
+        args = {
+            'backend': 'lava'
+        }
+        r = self.client.post('/api/submitjob/mygroup/myproject/1/myenv', args)
+        self.assertEqual(400, r.status_code)
+
     def test_accepts_definition_as_file_upload(self):
         args = {
             'backend': 'lava',

